### PR TITLE
[v5] Remove withAppProvider

### PR DESCRIPTION
**Pull request by BPScott**
### WHY are these changes introduced?

We're down to having a handful of components using withAppProvider. Let's make the final push to remove it.

### WHAT is this pull request doing?

- Convert Page to be a functional component
- Convert other usages of withAppProvider to use bespoke wrappers. These components are large and complex so a migration to being functional components will take time. This is a minimal approach, that is functionally very similar to withAppProvider but less convoluted. 

### How to 🎩

Confirm affected components look as expected in storybook

### CODE REVIEW

File: src/components/DataTable/DataTable.tsx [hunk 1]
```diff
@@ -441,4 +438,7 @@ class DataTableInner extends React.PureComponent<
   };
 }
 
-export const DataTable = withAppProvider<DataTableProps>()(DataTableInner);
+export function DataTable(props: DataTableProps) {
```
Comment (amrocha) [inline src/components/DataTable/DataTable.tsx:441]: Huh, why did you do it this way?

File: src/components/Page/Page.tsx [hunk 2]
```diff
@@ -18,46 +14,26 @@ export interface PageProps extends HeaderProps {
...
+
+  return (
+    <div className={className}>
+      {headerMarkup}
+      <div className={styles.Content}>{children}</div>
+    </div>
+  );
 }
-
-export const Page = withAppProvider<PageProps>()(PageInner);
```
Comment (amrocha) [inline src/components/Page/Page.tsx:63]: Was this even using AppProvider before?

**Comment by amrocha** [comment]
>These components are large and complex so a migration to being functional components will take time. This is a minimal approach, that is functionally very similar to withAppProvider but less convoluted.

Gotcha, makes sense to me and this lets us move forward and remove with-app-provider 👍

File: src/components/Page/Page.tsx (hunk 2 above)
Comment (BPScott) [reply src/components/Page/Page.tsx:63]: It was in v4. As part of ripping out app-bridge this component stopped using the `polaris` prop that withAppBridge provided, so yeah it not actually using what withAppProvider provided in the v5 branch.

I could have left this change as just removing the `withAppProvider` wrapping and having Page export a class component, but it was easy enough to migrate to a functional component so why not :)

**Pull request merged**